tests/sharding/run.sh

```diff
@@ -17,9 +17,8 @@ EOF
 }
 
 function run() {
-    run_sql "SET @@GLOBAL.SQL_MODE='NO_ZERO_IN_DATE,NO_ZERO_DATE'" $MYSQL_PORT1 $MYSQL_PASSWORD1
```

Comment:
in `db1.increment.sql` there's some zero datetime, so maybe this is a case to ensure without `STRICT_TRANS_TABLES` `NO_ZERO_DATE` doesn't block?

Reply:
Add it back for source1. Seems mariadb not allow zero_date even without `strict_trans_tables`